src/lib.rs

```diff
@@ -67,7 +68,14 @@ pub fn run_program_with_custom_state(bin_path: &str, mut vm: VMState) -> (U256,
                 Variant::Div(_) => _div(&mut vm, &opcode),
                 Variant::Jump(_) => todo!(),
                 Variant::Context(_) => todo!(),
-                Variant::Shift(_) => todo!(),
+                Variant::Shift(_) => _shift(
+                    &mut vm,
+                    &opcode,
+                    match opcode.variant {
+                        Variant::Shift(shift_type) => shift_type,
+                        _ => unreachable!(),
+                    },
+                ),
```

Comment:
Maybe we should expose `_shl`, `_shr`, `_rol` and `_ror` as public functions and match to them here, so we keep this `match` segment similar to `Variant::Ptr()` and `Variant::Log()`.

Reply:
Done! Thank you!!